sematic/plugins/building/docker_client_utils.py

```diff
@@ -0,0 +1,72 @@
+"""
+Utility code for interacting with the `docker-py` client.
+"""
+# Standard Library
+import sys
+import textwrap
+from typing import Any, Dict, Generator, List, Optional, TextIO
+
+_ROLLING_PRINT_COLUMNS = 120
+_ROLLING_PRINT_ROWS = 10
+
+
+def _rolling_print_status_updates(
+    status_updates: Generator[Dict[str, Any], None, None],
+    output_handle: TextIO = sys.stderr,
+    columns: int = _ROLLING_PRINT_COLUMNS,
+    rows: int = _ROLLING_PRINT_ROWS,
+) -> Optional[Dict[str, Any]]:
+    """
+    Prints a rolling window of Docker server status message updates.
+    """
+    message_window: List[str] = []
+
+    for status_update in status_updates:
+
+        if "error" in status_update.keys():
+            return status_update
+
+        update_str = _docker_status_update_to_str(status_update)
+        if update_str is None:
+            continue
+
+        line_count = len(message_window)
+        for raw_line in update_str.split("\n"):
+            wrapped_lines = textwrap.wrap(
+                raw_line, replace_whitespace=False, width=columns
+            )
+            message_window += wrapped_lines
+
+        if len(message_window) <= rows:
+            for message in message_window[line_count:]:
+                print(message, file=output_handle)
+
+        else:
+            message_window = message_window[-rows:]
+            message_window[0] = "[...]"
+
+            print(f"\033[{line_count + 1}F", file=output_handle)
```

Comment:
code might be more readable if you replace these escape sequences with constants:
```suggestion
            print(f"{_START_LINE_CLEAR}{line_count + 1}{_END_LINE_CLEAR}", file=output_handle)
```

Reply:
Added code comments instead.